mycore-oai/src/main/java/org/mycore/oai/MCROAIDeletedSearcher.java

```
@@ -0,0 +1,123 @@
+     * @param rows how many rows
+     * @return the cursor as string
+     */
+    public String buildCursor(int from, int rows) {
+        StringBuilder b = new StringBuilder(CURSOR_PREFIX);
+        b.append(CURSOR_DELIMETER);
+        b.append(from);
+        b.append(CURSOR_DELIMETER);
+        b.append(rows);
+        return b.toString();
+    }
+
+    /**
+     * Returns a list with identifiers of the deleted objects within the given date boundary.
+     * If the record policy indicates that there is no support for tracking deleted an empty
+     * list is returned.
+     * 
+     * @param from from date
+     * @param until to date
+     * 
+     * @return a list with identifiers of the deleted objects
+     */
+    protected List<String> searchDeleted(ZonedDateTime from, ZonedDateTime until) {
+        DeletedRecordPolicy deletedRecordPolicy = this.identify.getDeletedRecordPolicy();
+        if (from == null || DeletedRecordPolicy.No.equals(deletedRecordPolicy)
+            || DeletedRecordPolicy.Transient.equals(deletedRecordPolicy)) {
+            return new ArrayList<>();
+        }
+        LOGGER.info("Getting identifiers of deleted items");
+        try {
+            List<String> deletedItems = MCRDeletedItemManager.getDeletedItems(from, Optional.ofNullable(until));
+            List<String> types = getConfig().getStrings(getConfigPrefix() + "DeletedRecordTypes", null);
+            if (types == null) {
+                return deletedItems;
+            }
+            return deletedItems.stream()
+                               .map(MCRObjectID::getInstance)
+                               .filter(id -> {
+                                   return types.contains(id.getTypeId());
+                               })
+                               .map(MCRObjectID::toString)
+                               .collect(Collectors.toList());
+        } catch (Exception ex) {
+            LOGGER.warn("Could not retrieve identifiers of deleted objects", ex);
```

Error are not handled by logging them. Exceptions should be thrown until response code != 200 is given to the client.